### PR TITLE
gh-109015: Add test.support.socket_helper.tcp_blackhole()

### PR DESCRIPTION
**Pull request by vstinner**
* Issue: gh-109015

### CODE REVIEW

**Comment by vstinner** [comment]
My previous DEBUG PR: PR #108972.

See also issue #91960 "Add FreeBSD build and test using Cirrus-CI" and PR #91961 which the TCP blackhole was discussed.

**Comment by vstinner** [comment]
I validated my fix on Cirrus-CI with this patch:

```
diff --git a/.cirrus.yml b/.cirrus.yml
index 823b1f921d66d..2e087f1f57706 100644
--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -7,9 +7,9 @@ freebsd_task:
   # local local connections to fail with ETIMEDOUT instead of ECONNREFUSED.
   # For more information see https://reviews.freebsd.org/D41751 and
   # https://github.com/cirruslabs/cirrus-ci-docs/issues/483.
-  sysctl_script:
-    - sysctl net.inet.tcp.blackhole=0
-    - sysctl net.inet.udp.blackhole=0
+#  sysctl_script:
+#    - sysctl net.inet.tcp.blackhole=0
+#    - sysctl net.inet.udp.blackhole=0
   build_script:
     - mkdir build
     - cd build
```

test_asyncio failed at the first version of my PR, which proves that the Cirrus-CI did its job: test my fix :-) I had to skip 2 more tests in test_asyncio.

At the second version, it passed: ``Total test files: run=461/461 skipped=25 resource_denied=1 rerun=1``: https://cirrus-ci.com/task/4512145228955648 (test.test_asyncio.test_subprocess had to be re-run, it's unrelated to this fix.)

cc @emaste

**Comment by vstinner** [comment]
My second patch adds skips to Lib/test/test_asyncio/test_events.py:

* test_create_connection_local_addr_skip_different_family()
* test_create_connection_local_addr_nomatch_family()

**Comment by vstinner** [comment]
I reverted the Cirrus CI change.

File: Lib/test/support/socket_helper.py [hunk 1]
```diff
@@ -277,3 +278,62 @@ def create_unix_domain_name():
...
+_sysctl_cache = {}
+
+def _get_sysctl(name):
+    """Get a sysctl value as an integer."""
+    try:
+        return _sysctl_cache[name]
+    except KeyError:
+        pass
+
+    cmd = ['sysctl', name]
```
Comment (emaste) [inline Lib/test/support/socket_helper.py]: sysctl provides a `-n` arg to suppress the name:
```
$ sysctl -n net.inet.tcp.blackhole
0
```

Comment (vstinner) [reply Lib/test/support/socket_helper.py]: Interesting, I didn't know it. It's also supported by Linux (even if right now, this function is only called on FreeBSD ;-)). I updated my PR.

**Comment by vstinner** [comment]
I ran a quick test on my latest PR:

```
vstinner@freebsd$ sudo sysctl net.inet.tcp.blackhole=0
net.inet.tcp.blackhole: 0 -> 0
vstinner@freebsd$ ./python -m test -v test_socket -m test_connect -v
(...)
test_connect (test.test_socket.NetworkConnectionNoServer.test_connect) ... ok
(...)
Result: SUCCESS
```

and:

```
vstinner@freebsd$ sudo sysctl net.inet.tcp.blackhole=2
net.inet.tcp.blackhole: 0 -> 2

vstinner@freebsd$ ./python -m test -v test_socket -m test_connect -v
(...)
test_connect (test.test_socket.NetworkConnectionNoServer.test_connect) ... skipped 'TCP blackhole is enabled (sysctl 
(...)
Result: SUCCESS
```

**Comment by vstinner** [comment]
@emaste: Would you mind to review the updated PR?

File: Lib/test/support/socket_helper.py [hunk 2]
```diff
@@ -277,3 +278,62 @@ def create_unix_domain_name():
...
+                          text=True)
+    if proc.returncode:
+        support.print_warning(f'{' '.join(cmd)!r} command failed with '
+                              f'exit code {proc.returncode}')
+        # cache the error to only log the warning once
+        _sysctl_cache[name] = None
+        return None
+    output = proc.stdout
+
+    # Parse 'net.inet.tcp.blackhole: 0\n' to get '0'
```
Comment (emaste) [inline Lib/test/support/socket_helper.py]: I think this comment is leftover

**Review by emaste — APPROVED** [review]

**Review by emaste — APPROVED** [review]

**Pull request merged**